apps/src/templates/HideToolbarHelper.jsx

```diff
@@ -56,44 +46,50 @@ export default class HideToolbarHelper extends React.Component {
   }
 
   isToolbarShowing() {
-    // window.innerHeight is smaller than document.body.offsetHeight when
-    // the iOS 13 Safari toolbar is showing.  It becomes equal when the toolbar
-    // is hidden.  (Interestingly, document.documentElement.clientHeight is
-    // the same as document.body.offsetHeight in both cases.)
-    return window.innerHeight < document.body.offsetHeight;
+    // In general, window.innerHeight is smaller than document.body.offsetHeight
+    // and document.documentElement.clientHeight when the iOS 13 Safari toolbar
+    // is showing. However, as the page is loading, these three values change at
+    // different times when the viewport is loading. Therefore we check the
+    // innerHeight against the offsetHeight and the clientHeight. When the
+    // toolbar is hidden, all values are the same.
+    return (
+      window.innerHeight < document.body.offsetHeight ||
+      window.innerHeight < document.documentElement.clientHeight
+    );
   }
 
   updateLayout = () => {
     const isiOS13 = this.isiOS13();
     const isHideCookieSet = this.isHideCookieSet();
     const isLandscape = this.isLandscape();
 
-    if (!isiOS13 || !isLandscape || isHideCookieSet) {
-      this.setState({showHelper: false});
+    // Check if our environment is in a state where showing the helper would be
+    // useful. i.e. we are on iOS 13 (or have used the override), are in
+    // landscape, and we haven't hidden due to a cookie
+    if (
+      !((this.forceShowHelper || isiOS13) && isLandscape && !isHideCookieSet)
+    ) {
+      this.setState({shouldShowHelper: false});
       return;
     }
 
-    // We are on iOS 13, in landscape, and we haven't hidden due to a cookie.
     // Let's show the helper if we think the toolbar is showing.
-    const showHelper = this.isToolbarShowing();
+    const shouldShowHelper = this.isToolbarShowing() || this.forceShowHelper;
 
     // If we previously have shown the helper, and aren't now, and it's the
     // first time we've encountered this situation, then do a couple things.
-    if (this.wasHelperShowing && !showHelper && !this.didTrackToolbarHide) {
+    if (this.wasHelperShowing && !shouldShowHelper) {
       // Let's track the disapearance of the toolbar after we had previously
       // shown the helper.
       trackEvent('Research', 'HideToolbarHelper', 'hid-' + window.innerHeight);
 
       // And also set the cookie so this user doesn't see the helper again
       // for a year.
       this.setHideHelperCookie();
-
-      this.didTrackToolbarHide = true;
```

Comment:
This was used to avoid writing the GA event more than once per page viewing.  Do we still protect against that in all cases?